.github/workflows/build_deploy.yaml

```diff
@@ -0,0 +1,148 @@
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10' # Updated to Python 3.10
+
+    - name: Install system dependencies
+      run: sudo apt-get install jq
+
+    - name: Read Checksum from config.env
+      run: |
+        echo "Reading checksum..."
+        CHECKSUM=$(cat config.env | grep REPROSCHEMA_UI_CHECKSUM | cut -d '=' -f2)
+        echo "LATEST_COMMIT_HASH=${CHECKSUM}" >> $GITHUB_ENV
+
+    - name: Install Node dependencies and validate JSON
+      run: |
+        npm install -g jsonlint
+        # Validate JSON files for both protocols
+        find . -name "*.json" -path "./peds-protocol-*" | xargs -I {} jsonlint -q {}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools
+        pip install reproschema requests_cache pre-commit
+    
+    - name: Test with pyshacl
+      run: |
+        # Validate both protocol schemas
+        reproschema -l DEBUG validate ${PROTOCOL_COMBINED_SLUG}/activities
+        reproschema -l DEBUG validate ${PROTOCOL_COMBINED_SLUG}/peds-protocol/peds-protocol
+        reproschema -l DEBUG validate ${PROTOCOL_QUESTIONNAIRES_SLUG}/activities
+        reproschema -l DEBUG validate ${PROTOCOL_QUESTIONNAIRES_SLUG}/peds-protocol/peds-protocol
+
+    - name: Build Combined Protocol UI
+      run: |
+        # Download and setup UI for combined protocol
+        curl -O -sSL https://github.com/ReproNim/reproschema-ui/archive/${LATEST_COMMIT_HASH}.zip
+        unzip ${LATEST_COMMIT_HASH}.zip
+        mv reproschema-ui-* ui-combined
+        cd ui-combined
+        cp -r ../${PROTOCOL_COMBINED_SLUG}/ui-changes/. .
+        # Modify the title in index.html
+        sed -i 's/<title>.*<\/title>/<title>B2AI Peds Protocol - Combined<\/title>/' public/index.html
+        npm install
+        npm run build
+        cd ..
+
+    - name: Build Questionnaires Protocol UI
+      run: |
+        # Download and setup UI for questionnaires protocol
+        curl -O -sSL https://github.com/ReproNim/reproschema-ui/archive/${LATEST_COMMIT_HASH}.zip
+        unzip ${LATEST_COMMIT_HASH}.zip
+        mv reproschema-ui-* ui-questionnaires
+        cd ui-questionnaires
+        cp -r ../${PROTOCOL_QUESTIONNAIRES_SLUG}/ui-changes/. .
+        # Modify the title in index.html
+        sed -i 's/<title>.*<\/title>/<title>B2AI Peds Protocol - Questionnaires<\/title>/' public/index.html
+        npm install
+        npm run build
+        cd ..
+
+    - name: Prepare deployment folder
+      run: |
+        # Create deployment folder structure
+        mkdir -p deploy
+        mkdir -p deploy/combined
+        mkdir -p deploy/questionnaires
+        
+        # Copy built UIs to respective folders
+        cp -r ui-combined/dist/* deploy/combined/
+        cp -r ui-questionnaires/dist/* deploy/questionnaires/
+        
+        # Create root index.html to redirect or show options
+        cat > deploy/index.html << 'EOF'
```

Comment:
if this index.html does not depend on any variables in the github action (which it doesn't seem too), then it'll be clearer to add it directly to the repo as well and cp it to the correct place or something similar

Reply:
removed it outside.